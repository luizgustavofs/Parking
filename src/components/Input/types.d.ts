export interface InputPlaceholderProps {
  color?: string;
}

export interface InputProps extends InputPlaceholderProps {
  placeholderText: string;
  label: string;
  onChange: (value: string) => void;
  value: string;
}
