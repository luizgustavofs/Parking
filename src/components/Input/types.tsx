export interface InputPlaceholderProps {
  color?: string;
}

export interface InputProps extends InputPlaceholderProps {
  placeholderText: string;
  label: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  value?: string;
}
