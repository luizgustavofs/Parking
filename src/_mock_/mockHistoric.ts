export const MOCK_HISTORIC_EMPTY = [];

export const MOCK_HISTORIC_WITHOUT_LEFT = [
  {
    time: '12 days 5 hours',
    paid: true,
    left: false,
    plate: 'AAA-1111',
    reservation: '607ecf',
  },
  {
    time: '37 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d8',
  },
  {
    time: '20 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '60451e',
  },
  {
    time: '12 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6047d2',
  },
  {
    time: '52 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
  {
    time: '33 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
];

export const MOCK_HISTORIC_WITHOUT_PAYMENT = [
  {
    time: '12 days 5 hours',
    paid: false,
    left: false,
    plate: 'AAA-1111',
    reservation: '607ecf',
  },
  {
    time: '37 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d8',
  },
  {
    time: '20 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '60451e',
  },
  {
    time: '12 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6047d2',
  },
  {
    time: '52 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
  {
    time: '33 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
];

export const MOCK_HISTORIC_COMPLETE = [
  {
    time: '12 days 5 hours',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '607ecf',
  },
  {
    time: '37 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d8',
  },
  {
    time: '20 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '60451e',
  },
  {
    time: '12 minutes',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6047d2',
  },
  {
    time: '52 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
  {
    time: '33 seconds',
    paid: true,
    left: true,
    plate: 'AAA-1111',
    reservation: '6046d9',
  },
];
