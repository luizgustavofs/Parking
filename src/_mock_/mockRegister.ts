export const MOCK_REGISTER_OK = {
  reservation: '62733f',
  plate: 'AAA-0044',
  entered_at: '2022-05-05T03:07:50.710+00:00',
};
