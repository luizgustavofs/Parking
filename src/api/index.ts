import axios from 'axios';

export const api = axios.create({
  baseURL: 'https://parking-lot-to-pfz.herokuapp.com',
  timeout: 5000,
  headers: {},
});
