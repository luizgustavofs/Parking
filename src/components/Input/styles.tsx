import styled from 'styled-components';

import { InputPlaceholderProps } from './types';

export const Container = styled.div<InputPlaceholderProps>`
  width: 100%;
  flex-direction: column;

  h3 {
    color: ${({ theme }) => theme.colors.neutral._80};
    margin-bottom: 7px;
  }

  input {
    height: 67px;
    width: 100%;
    text-align: center;
    background: ${({ theme }) => theme.colors.baseColor};
    border: solid 1px ${({ theme }) => theme.colors.neutral._70};
    color: ${({ color }) => color};
    margin-bottom: 13px;
  }
`;
