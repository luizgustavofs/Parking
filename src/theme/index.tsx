import { ReactNode } from 'react';

import { ThemeProvider } from 'styled-components';

import { colors } from './colors';
import { fontSizes } from './fontSizes';
import { spaces } from './spaces';

export const theme = {
  colors,
  spaces,
  fontSizes,
};

const ProjectThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
  ...props
}) => {
  return (
    <ThemeProvider theme={theme} {...props}>
      {children}
    </ThemeProvider>
  );
};

export { ProjectThemeProvider };
