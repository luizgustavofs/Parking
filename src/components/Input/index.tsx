import React from 'react';

import InputMask from 'react-input-mask';
import { useTheme } from 'styled-components';

import { Container } from './styles';
import { InputProps } from './types';

const Input: React.FC<InputProps> = ({
  placeholderText,
  label,
  color,
  value,
  onChange,
}) => {
  const { colors } = useTheme();

  return (
    <>
      <Container color={color || colors.neutral._80}>
        <h3>{label}</h3>
        <InputMask
          mask="aaa-9*99"
          placeholder={placeholderText}
          onChange={onChange}
          value={value}
        />
      </Container>
    </>
  );
};

export default Input;
