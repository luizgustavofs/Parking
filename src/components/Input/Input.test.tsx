import { fireEvent } from '@testing-library/dom';
import { render } from '@testing-library/react';
import AxiosMockAdapter from 'axios-mock-adapter';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Input from '.';
import { MOCK_HISTORIC_COMPLETE } from '../../_mock_/mockHistoric';
import { api } from '../../api';
import { theme } from '../../theme';

describe('Input', () => {
  it('should render correctly with variants', () => {
    const { toJSON } = create(
      <ThemeProvider theme={theme}>
        <Input
          label="teste"
          placeholderText="test"
          onChange={() => {}}
          value="teste"
        />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should change de value input if change plate', () => {
    var mock = new AxiosMockAdapter(api);

    mock.onGet('/parking/AAA-0000').reply(200, MOCK_HISTORIC_COMPLETE);
    mock.onPost('/parking/AAA-0000').reply(200, {});

    const changePlateMock = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Input
          label="teste"
          placeholderText="test"
          onChange={changePlateMock}
          value="teste"
        />
      </ThemeProvider>,
    );

    const input = getByTestId('input-test');
    fireEvent.change(input, { target: { value: 'NEW-0000' } });

    expect(changePlateMock).toHaveBeenCalledWith('NEW-0000');
  });
});
