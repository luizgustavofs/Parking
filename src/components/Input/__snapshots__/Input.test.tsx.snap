// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly with variants 1`] = `
<div
  className="sc-bczRLJ kuhJBS"
  color="#9B9B9B"
>
  <h3>
    teste
  </h3>
  <input
    data-testid="input-test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onPaste={[Function]}
    placeholder="test"
    value="tes-____"
  />
</div>
`;
